plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/IcebergNessieCatalogModule.java

```diff
@@ -58,9 +63,46 @@ public static NessieIcebergClient createNessieIcebergClient(IcebergNessieCatalog
         icebergNessieCatalogConfig.getBearerToken()
                 .ifPresent(token -> builder.withAuthentication(BearerAuthenticationProvider.create(token)));
 
-        return new NessieIcebergClient(builder.build(NessieApiV1.class),
+        // default version is inferred by uri.
+        IcebergNessieCatalogConfig.ClientApiVersion apiVersion = icebergNessieCatalogConfig.getClientAPIVersion()
```

Comment:
let's push this logic to IcebergNessieCatalogConfig so that getClientAPIVersion always return a `ClientApiVersion`:

```
    public ClientApiVersion getClientAPIVersion()
    {
        return clientAPIVersion.orElseGet(this::inferVersionFromURI);
    }
```

Move `inferVersionFromURI` to `IcebergNessieCatalogConfig`

Reply:
`TestIcebergNessieCatalogConfig.testExplicitPropertyMapping` is failing because `assertAttributesNotEqual(metadata, actual, defaultInstance);` will try to get the `getClientAPIVersion()` using getter method of the default instance (without URI and get the exception). 

I think clubbing `inferVersionFromURI` in the getter is not a good idea. I am planning to call it separately but still keep the `inferVersionFromURI` method in this class as static method.